#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_depth_change_factor", double_t, 0, "max depth change factor", 0.1, 0, 1)
gen.add("normal_smoothing_size", double_t, 0, "normal smoothing size", 1, 0, 10)
gen.add("radius_search_knn", double_t, 0, "radius search k-NN", 0.05, 0.01, 1)
normal_estimation_mode_enum = gen.enum([
    gen.const("ORGANIZED_FAST_NORMALS", int_t, 0, "ORGANIZED_FAST_NORMALS"),
    gen.const("COVARIANCE_MATRIX", int_t, 1, "COVARIANCE_MATRIX"),
    gen.const("AVERAGE_3D_GRADIENT", int_t, 2, "AVERAGE_3D_GRADIENT"),
    gen.const("AVERAGE_DEPTH_CHANGE", int_t, 3, "AVERAGE_DEPTH_CHANGE"),
    gen.const("SIMPLE_3D_GRADIENT", int_t, 4, "SIMPLE_3D_GRADIENT"),
    gen.const("STANDARD_KNN_NORMALS", int_t, 5, "STANDARD_KNN_NORMALS")],
    "Normal Estimation Mode")

gen.add("normal_estimation_mode", int_t, 0, "Normal Estimation Mode", 0, 0, 5, edit_method=normal_estimation_mode_enum)

exit(gen.generate("organized_point_normals", "organized_point_normals", "OrganizedPointNormals"))
